src/Uno.UWP/Storage/StorageFolder.Local.cs

```diff
@@ -1,96 +1,268 @@
-﻿#pragma warning disable CS1998 // Async method lacks 'await' operators and will run synchronously
+﻿#nullable enable
+#pragma warning disable CS1998 // Async method lacks 'await' operators and will run synchronously
 
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Uno.Extensions;
+using Windows.Storage.FileProperties;
+using Windows.Storage.Internal;
+using IOPath = global::System.IO.Path;
 
 #if __IOS__
 using UIKit;
 using Foundation;
 #endif
 
-
-using System;
-using System.IO;
-using Uno.Extensions;
-using Windows.Foundation;
-
 namespace Windows.Storage
 {
 	partial class StorageFolder
 	{
+		internal StorageFolder(string fullPath)
+			: this(new Local(IOPath.GetFileName(fullPath), fullPath))
+		{
+		}
+
+		internal StorageFolder(string name, string path)
+			: this(new Local(name, path))
+		{
+		}
+
 		private sealed class Local : ImplementationBase
 		{
-			public Local(string path)
+			private static readonly StorageProvider _provider = new StorageProvider("computer", "This PC");
```

Comment:
Is `"This PC"` shown to end user? Maybe this could need to be translatable.

Reply:
The "computer" is an identifier only, but "This PC" can be localized 👍